hr_contract_amendment/models/hr_contract.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+#   Copyright (C) 2018 EBII (https://www.saaslys.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, api
+
+
+class HrContract(models.Model):
+    _inherit = 'hr.contract'
+
+    amendment_ids = fields.One2many(
+        comodel_name='hr.contract.amendment', inverse_name="contract_id",
+        string="contract", required=False)
+
+    @api.multi
+    def write(self, vals):
```

Comment:
i see this algorithm a little complex. (I mean Overloading write function to make such changes.
What about 
* set job_id, wage as readonly on hr.contract
* create a wizard to allow to change the values

Reply:
it is automatic process to follow change on contract.

keep each change between you create it and the time you look in is saved to know what about changes.
Who , when which values before and after ..  without one boring mouse click for user. 
where add this either than on write method ?